types/mapcmd/map_it_test.go

```diff
@@ -0,0 +1,448 @@
+		tcs := []struct {
+			name        string
+			args        string
+			key         interface{}
+			value       interface{}
+			errContains string
+		}{
+			{
+				name:  "valid remove key(string)",
+				args:  "--key k1" + mapNameArg,
+				key:   "k1",
+				value: "v1",
+			},
+			{
+				name:  "valid remove key(json)",
+				args:  `--key {"some":"key"} --key-type json` + mapNameArg,
+				key:   serialization.JSON(`{"some":"key"}`),
+				value: serialization.JSON(`{"some":"value"}`),
+			},
+			{
+				name:        "invalid remove, missing key",
+				args:        "" + mapNameArg,
+				errContains: `"key" not set`,
+			},
+			{
+				name:        "invalid remove, missing map name",
+				args:        "--key k1",
+				errContains: `"name" not set`,
+			},
+		}
+		for _, tc := range tcs {
+			t.Run(tc.name, func(t *testing.T) {
+				cmd := NewRemove(c)
+				var stdout, stderr bytes.Buffer
+				cmd.SetOut(&stdout)
+				cmd.SetErr(&stderr)
+				args := strings.Split(tc.args, " ")
+				cmd.SetArgs(args)
+				ctx := context.Background()
+				if tc.key != nil && tc.value != nil {
+					require.NoError(t, m.Set(ctx, tc.key, tc.value))
+				}
+				_, err := cmd.ExecuteContextC(ctx)
+				if tc.errContains != "" {
+					require.Contains(t, err.Error(), tc.errContains)
+					return
+				}
+				require.NoError(t, err)
+				// cmd should not print anything
+				require.Empty(t, stdout.String())
+				require.Empty(t, stderr.String())
+				ok, err := m.ContainsKey(ctx, tc.key)
+				require.NoError(t, err)
+				require.False(t, ok)
+			})
+		}
+	})
+}
+
+func TestMapClear(t *testing.T) {
+	it.MapTesterWithConfigAndMapName(t, func(t *testing.T, c *hazelcast.Config, m *hazelcast.Map, mn string) {
+		//populate map
+		key1, key2 := "k1", "k2"
+		entries := []types.Entry{
+			{
+				Key:   key1,
+				Value: "test",
+			},
+			{
+				Key:   key2,
+				Value: "test",
+			},
+		}
+		require.NoError(t, m.PutAll(context.Background(), entries...))
+		mapNameArg := fmt.Sprintf(` --name %s`, mn)
+		tcs := []struct {
+			name        string
+			args        string
+			errContains string
+		}{
+			{
+				name:        "invalid clear, missing map name",
+				args:        "",
+				errContains: `"name" not set`,
+			},
+			{
+				name: "clear removes the entries",
+				args: "" + mapNameArg,
+			},
+		}
+		for _, tc := range tcs {
+			t.Run(tc.name, func(t *testing.T) {
+				cmd := NewClear(c)
+				var stdout, stderr bytes.Buffer
+				cmd.SetOut(&stdout)
+				cmd.SetErr(&stderr)
+				args := strings.Split(tc.args, " ")
+				cmd.SetArgs(args)
+				ctx := context.Background()
+				_, err := cmd.ExecuteContextC(ctx)
+				if tc.errContains != "" {
+					require.Contains(t, err.Error(), tc.errContains)
+					return
+				}
+				require.NoError(t, err)
+				// cmd should not print anything
+				require.Empty(t, stdout.String())
+				require.Empty(t, stderr.String())
+				size, err := m.Size(ctx)
+				require.NoError(t, err)
+				if size != 0 {
+					t.FailNow()
```

Comment:
Would it make sense to fail with `t.Fatalf("reason")`?

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/114/commits/33e5d0a48d545a710b929043f671ab0808485bcf